java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryHashTable.java

```diff
@@ -95,6 +97,18 @@ public DictionaryHashTable(ValueVector dictionary) {
     this(DEFAULT_INITIAL_CAPACITY, dictionary);
   }
 
+  /**
+   * Build hash table with dictionary vector.
+   */
+  public void ensureBuildTable() {
```

Comment:
why not put this in the constructor either in this class or in DictionaryEncoder?

Reply:
You are right, fixe now, thanks.